### PR TITLE
Add Vapor Getting Started guide

### PR DESCRIPTION
**Pull request by 0xTim**
Adds the getting started guide for Vapor

### CODE REVIEW

File: getting-started/vapor-web-server/index.md [hunk 1]
```diff
@@ -3,4 +3,152 @@ layout: page
...
+```bash
+$ curl http://localhost:8080/hello/tim
+Hello, Tim!
+```
+
+Try it with different names to see it change automatically!
+
+## Returning JSON
+
+Vapor uses `Codable` under the hood to make it easy to send and receive JSON, using a wrapper protocol called `Content` to add a few extra features. Next, you'll return a JSON body with the message from the Hello! route. First, create a new type at the bottom of **routes.swift**:
```
Comment (MaxDesiatov) [inline getting-started/vapor-web-server/index.md:77]: Would you consider linking from this mention of `Codable` to `Codable` reference page?

Comment (0xTim) [reply getting-started/vapor-web-server/index.md:77]: Sure, you mean https://developer.apple.com/documentation/swift/codable?

Comment (MaxDesiatov) [reply getting-started/vapor-web-server/index.md:77]: Yes, if you think that's appropriate in this context.

Comment (ktoso) [reply getting-started/vapor-web-server/index.md:77]: That sounds like a good idea 👍

File: getting-started/vapor-web-server/index.md [hunk 2]
```diff
@@ -3,4 +3,152 @@ layout: page
...
+```
+
+For Xcode, run:
+
+```bash
+cd HelloVapor
+open Package.swift
+```
+
+Vapor's template contains a number of files and functions already set up for you. **configure.swift** contains the code to configure your application and **routes.swift** contains route handler code.
```
Comment (ktoso) [inline getting-started/vapor-web-server/index.md:44]: maybe a bit nitpicky, but filenames I'd expect in `` code blocks, not "**bold**"

**Review by ktoso — APPROVED** [review]
Looks good to me :)

File: getting-started/vapor-web-server/index.md (hunk 1 above)
Comment (0xTim) [reply getting-started/vapor-web-server/index.md]: Done!

File: getting-started/vapor-web-server/index.md [hunk 3]
```diff
@@ -3,4 +3,152 @@ layout: page
...
+```bash
+$ curl http://localhost:8080/hello/tim
+Hello, Tim!
+```
+
+Try it with different names to see it change automatically!
+
+## Returning JSON
+
+Vapor uses [`Codable`](https://developer.apple.com/documentation/swift/codable) under the hood to make it easy to send and receive JSON, using a wrapper protocol called `Content` to add a few extra features. Next, you'll return a JSON body with the message from the Hello! route. First, create a new type at the bottom of **routes.swift**:
```
Comment (MaxDesiatov) [inline getting-started/vapor-web-server/index.md:77]: Actually, just realized that page doesn't have good examples of use of `Codable`, just two abstract description sentences. TSPL doesn't have a tutorial for use of `Codable` either. Do we have any good introductions to `Codable` that we could link from here?

Comment (0xTim) [reply getting-started/vapor-web-server/index.md:77]: The best one I know is https://www.youtube.com/watch?v=yL5Ff5p1hyc

File: getting-started/vapor-web-server/index.md [hunk 4]
```diff
@@ -3,4 +3,152 @@ layout: page
 title: Build a Web Server
```
Comment (tomerd) [inline getting-started/vapor-web-server/index.md:3]: Build a Web Server with Vapor web framework?

File: getting-started/vapor-web-server/index.md [hunk 5]
```diff
@@ -3,4 +3,152 @@ layout: page
...
 ---
 
-TODO
\ No newline at end of file
+{% include getting-started/_installing.md %}
+
+## Installing Vapor
```
Comment (tomerd) [inline getting-started/vapor-web-server/index.md:8]: is this required or optional?

File: getting-started/vapor-web-server/index.md (hunk 3 above)
Comment (tomerd) [reply getting-started/vapor-web-server/index.md:77]: cc @TimTr  @parkera for the missing Coddle documentation aspect

File: getting-started/vapor-web-server/index.md [hunk 6]
```diff
@@ -3,4 +3,152 @@ layout: page
...
+}
+```
+
+This defines a new type that conforms to `Content` that matches the JSON you want to return.
+
+Create a new route below `app.get("hello", ":name") { ... }` to return this JSON:
+
+```swift
+// 1
+app.get("json", ":name") { req async throws -> UserResponse in
```
Comment (tomerd) [inline getting-started/vapor-web-server/index.md:91]: nit `request` instead of `req` more aligned with api guidelines?

**Review by tomerd — APPROVED** [review]

File: getting-started/vapor-web-server/index.md (hunk 3 above)
Comment (MaxDesiatov) [reply getting-started/vapor-web-server/index.md:77]: Having done a bit more searching, there's a good intro at https://developer.apple.com/documentation/foundation/archives_and_serialization/encoding_and_decoding_custom_types, it just doesn't appear high in search results via `Codable` search term in the reference.

Comment (MaxDesiatov) [inline getting-started/vapor-web-server/index.md:77]:
```suggestion
Vapor uses [`Codable`](https://developer.apple.com/documentation/foundation/archives_and_serialization/encoding_and_decoding_custom_types) under the hood to make it easy to send and receive JSON, using a wrapper protocol called `Content` to add a few extra features. Next, you'll return a JSON body with the message from the Hello! route. First, create a new type at the bottom of **routes.swift**:
```

File: getting-started/vapor-web-server/index.md [hunk 7]
```diff
@@ -3,4 +3,152 @@ layout: page
...
+
+Then, in your terminal in a directory where you want to create the new project run:
+
+```bash
+vapor new HelloVapor
+```
+
+This pulls down a template and asks you a series of questions to create a simple project with everything you need to get started. This guide will create a simple REST API that you can send and receive JSON to and from. So answer no to all other questions. You'll see the project created successfully:
+
+![A New Vapor Project]({{site.url}}/assets/images/getting-started-guides/web-server/new-project.png)
```
Comment (tomerd) [inline getting-started/vapor-web-server/index.md:28]: web-server -> web-server-vapor (or similar more qualified path)

File: getting-started/vapor-web-server/index.md [hunk 8]
```diff
@@ -3,4 +3,152 @@ layout: page
...
+```bash
+cd HelloVapor
+code .
+```
+
+For Xcode, run:
+
+```bash
+cd HelloVapor
+open Package.swift
```
Comment (tomerd) [inline getting-started/vapor-web-server/index.md:41]: `xed .` also works

File: getting-started/vapor-web-server/index.md (hunk 5 above)
Comment (0xTim) [reply getting-started/vapor-web-server/index.md:8]: Required to get the template set up for you

File: getting-started/vapor-web-server/index.md (hunk 6 above)
Comment (0xTim) [reply getting-started/vapor-web-server/index.md:91]: Easy either way, all the Vapor docs and examples use `req`

**Review by MaxDesiatov — APPROVED** [review]

**Pull request merged**